app/javascript/Components/Result/marks_panel.jsx

```diff
@@ -13,22 +13,45 @@ export class MarksPanel extends React.Component {
     super(props);
     this.state = {
       expanded: new Set(),
+      activeCriterionId: null,
     };
   }
 
+  setActiveCriterion = id => {
+    this.setState({activeCriterionId: id});
+  };
+
+  nextCriterion = () => {
+    const criteria = this.props.marks.filter(
+      c => !(this.props.assigned_criteria !== null && !this.props.assigned_criteria.includes(c.id))
+    ); // skip unassigned
+
+    const idx = criteria.findIndex(c => c.id === this.state.activeCriterionId);
+    const next = idx === -1 || idx === criteria.length - 1 ? criteria[0] : criteria[idx + 1];
+    this.setActiveCriterion(next.id);
+  };
+
+  prevCriterion = () => {
+    const criteria = this.props.marks.filter(
+      c => !(this.props.assigned_criteria !== null && !this.props.assigned_criteria.includes(c.id))
+    );
+
+    const idx = criteria.findIndex(c => c.id === this.state.activeCriterionId);
+    const prev = idx <= 0 ? criteria[criteria.length - 1] : criteria[idx - 1];
+    this.setActiveCriterion(prev.id);
+  };
+
   componentDidMount() {
     if (!this.props.released_to_students) {
-      // TODO: Convert this to pure React
-      // Capture the mouse event to add "active-criterion" to the clicked element
-      $(document).on("click", ".rubric_criterion, .flexible_criterion, .checkbox_criterion", e => {
-        let criterion = $(e.target).closest(
-          ".rubric_criterion, .flexible_criterion, .checkbox_criterion"
-        );
-        if (!criterion.hasClass("unassigned")) {
-          e.preventDefault();
-          activeCriterion(criterion);
-        }
-      });
+      // Set first criterion active by default
```

Comment:
This logic can actually be moved to the constructor, since setting the state value only depends on props values.